source/dub/generators/generator.d

```diff
@@ -436,19 +437,50 @@ private void finalizeGeneration(string pack, in BuildSettings buildsettings, Pat
 	}
 }
 
-void runBuildCommands(in string[] commands, in BuildSettings build_settings)
+void runBuildCommands(in string[] commands, in Package pack, in GeneratorSettings settings, in BuildSettings build_settings)
 {
+	import std.conv;
 	import std.process;
 	import dub.internal.utils;
 
 	string[string] env = environment.toAA();
 	// TODO: do more elaborate things here
 	// TODO: escape/quote individual items appropriately
-	env["DFLAGS"] = join(cast(string[])build_settings.dflags, " ");
-	env["LFLAGS"] = join(cast(string[])build_settings.lflags," ");
-	env["VERSIONS"] = join(cast(string[])build_settings.versions," ");
-	env["LIBS"] = join(cast(string[])build_settings.libs," ");
-	env["IMPORT_PATHS"] = join(cast(string[])build_settings.importPaths," ");
-	env["STRING_IMPORT_PATHS"] = join(cast(string[])build_settings.stringImportPaths," ");
+	env["DFLAGS"]                = join(cast(string[])build_settings.dflags, " ");
+	env["LFLAGS"]                = join(cast(string[])build_settings.lflags," ");
+	env["VERSIONS"]              = join(cast(string[])build_settings.versions," ");
+	env["LIBS"]                  = join(cast(string[])build_settings.libs," ");
+	env["IMPORT_PATHS"]          = join(cast(string[])build_settings.importPaths," ");
+	env["STRING_IMPORT_PATHS"]   = join(cast(string[])build_settings.stringImportPaths," ");
+
+	env["DC"]                    = settings.platform.compilerBinary;
+	env["DC_BASE"]               = settings.platform.compiler;
+	env["D_FRONTEND_VER"]        = to!string(settings.platform.frontendVersion);
+
+	env["DUB_PLATFORM"]          = join(cast(string[])settings.platform.platform,"-");
+	env["DUB_ARCH"]              = join(cast(string[])settings.platform.architecture,"-");
```

Comment:
`" "` might make more sense as a delimiter for these two. Otherwise it may be confusing because of its similarity to the platform suffix syntax (`<platform>[-<arch>[-<compiler>]]`).